src/libcadet/model/ModelSystemImpl.cpp

```diff
@@ -278,12 +278,15 @@ unsigned int ModelSystem::numPureDofs() const CADET_NOEXCEPT
 	return dofs;
 }
 
-unsigned int const* ModelSystem::getModelStateOffsets() const CADET_NOEXCEPT
+std::tuple<unsigned int, unsigned int> ModelSystem::getModelStateOffsets(UnitOpIdx unitOp) const CADET_NOEXCEPT
 {
-	return _dofOffset.data();
+	for (IUnitOperation* m : _models)
+	{
+		if (m->unitOperationId() == unitOp)
+			return  std::make_tuple(_dofOffset[(unsigned int)m->unitOperationId()], _dofOffset[(unsigned int)m->unitOperationId()+1]);
+	}
```

Comment:
I have changed the function and now returning a tuple instead with _dofOffset[index] and _dofOffset[index+1].
I have compiled and checked the outputs and getting the right results.
Will that approach work?